packages/office-ui-fabric-react/src/components/Button/BaseButton.tsx

```diff
@@ -639,6 +639,9 @@ export class BaseButton extends BaseComponent<IBaseButtonProps, IBaseButtonState
       ev.preventDefault();
       ev.stopPropagation();
     }
+
+    // Note: When enter is pressed, we will let the event continue to propagate
+    // to trigger the onClick event on the button
```

Comment:
This comment seems a little out of place since there is no code under it (I get what you are trying to do here but it feels weird)... consider moving it to a better place